### PR TITLE
Nft base model

### PR DESCRIPTION
**Pull request by ruseinov**

### CODE REVIEW

File: x/nft/codec.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    repeated ApprovalSet approvalSet = 3;
+}
+
+message ApprovalSet {
+    string action = 1;
+    repeated Approval approvals = 2;
+}
+
+message Approval {
+    bytes toAccount = 1;
```
Comment (ruseinov) [inline x/nft/codec.proto]: I think this has to be changed to `address` or something, a bit cleaner naming. As there is only one account involved - no need for any additional explanation

File: x/nft/codec.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+message SetApprovalMsg {
+    bytes id = 1; // token id
+    string action = 2;
+    repeated Approval approvals = 3;
+}
+
+message RevokeApprovalMsg {
+    bytes id = 1; // token id
+    bytes toAccount = 2;
```
Comment (ruseinov) [inline x/nft/codec.proto]: same, shall we get rid of toAccount in favour of just account or address?

File: x/nft/codec.proto [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+package nft;
+
+message NonFungibleToken {
+    bytes id = 1;
+    bytes owner = 2;
+    repeated ApprovalSet approvalSet = 3;
+}
+
+message ApprovalSet {
```
Comment (ruseinov) [inline x/nft/codec.proto]: While I like this name it does not reflect the contents, as the content is an object with 2 fields.
I would still settle for ActionApprovals, even though I don't like that name as much it's much more clear

File: x/nft/username/codec.proto [hunk 4]
```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package username;
+
+import "github.com/iov-one/weave/x/nft/codec.proto";
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+
+message UsernameToken {
```
Comment (ruseinov) [inline x/nft/username/codec.proto]: We are assuming that the username is stored within the id, right?
Makes sense, as it should be unique in this case.

File: x/nft/codec.proto (hunk 1 above)
Comment (alpe) [reply x/nft/codec.proto]: yes, it should be address kind. good point.

File: x/nft/username/codec.proto (hunk 4 above)
Comment (alpe) [reply x/nft/username/codec.proto]: yes, Id = name

File: x/nft/codec.proto (hunk 2 above)
Comment (alpe) [reply x/nft/codec.proto]: ja

File: x/nft/codec.proto (hunk 3 above)
Comment (alpe) [reply x/nft/codec.proto]: q: IMHO it is a Set type but how about `ApprovedActions`?

Comment (alpe) [reply x/nft/codec.proto]: btw. you are right with the Set

**Comment by alpe** [comment]
It is a bit odd to be able to approve my own code like this. I am going to open a new PR from the branch.

**Pull request closed without merging**